pkg/interfaces/contracts/pool-cow/ICowRouter.sol

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { SwapKind } from "../vault/VaultTypes.sol";
+
+interface ICowRouter {
+    /**
+     * @notice Data for the swap and donate hook.
+     * @dev Swap and donate hook is used to swap on CoW AMM pools and donate surplus or fees to the same pool.
+     * @param pool Address of the CoW AMM Pool
+     * @param sender Account originating the swap and donate operation
+     * @param swapKind Type of swap (exact in or exact out)
+     * @param swapTokenIn Token to be swapped from
```

Comment:
Consider using the "token entering/leaving the Vault" terminology used below for `CoWSwapAndDonation`.